resolwe_bio/tests/unit/test_presample.py

```diff
@@ -0,0 +1,57 @@
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import unittest
+
+from django.contrib.auth import get_user_model
+
+from rest_framework.test import APIRequestFactory, force_authenticate
+
+from resolwe_bio.models import Sample
+from resolwe_bio.views import SampleViewSet, PresampleViewSet
+
+
+class PresampleTestCase(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls.user = get_user_model().objects.create(username='user')
+
+        sample_viewset = SampleViewSet()
+        cls.sample_queryset = sample_viewset.get_queryset()
+        presample_viewset = PresampleViewSet()
+        cls.presample_queryset = presample_viewset.get_queryset()
+
+        url_mapping = {
+            'get': 'retrieve',
+            'post': 'create',
+            'put': 'update',
+            'patch': 'partial_update',
+            'delete': 'destroy',
+        }
+
+        cls.sample_view = SampleViewSet.as_view(url_mapping)
+        cls.presample_view = PresampleViewSet.as_view(url_mapping)
+
+        cls.factory = APIRequestFactory()
+
+    @classmethod
+    def tearDownClass(cls):
+        Sample.objects.all().delete()
+        cls.user.delete()
+
+    def test_querysets(self):
+        sample = Sample.objects.create(contributor=self.user)
+        self.assertEqual(self.sample_queryset.count(), 0)
+        self.assertEqual(self.presample_queryset.count(), 1)
+
+        sample.presample = False  # mark sample as annotated
+        sample.save()
+        self.assertEqual(self.sample_queryset.count(), 1)
+        self.assertEqual(self.presample_queryset.count(), 0)
+
+    def test_update(self):
+        sample = Sample.objects.create(contributor=self.user)
+
+        request = self.factory.patch(self.presample_view, data={}, format='json')
```

Comment:
I'm working on this test, this is last part that is missing. I think that requests will be ok, as presample is still part of the queryset at the moment that request is made.